code/controllers/subsystems/supply.dm

```diff
@@ -191,12 +182,42 @@ SUBSYSTEM_DEF(supply)
 							if(!(dish.virus2.uniqueID in sold_virus_strains))
 								add_points_from_source(5, "virology_dishes")
 								sold_virus_strains += dish.virus2.uniqueID
+						qdel(dish)
+						continue
+
+					//Sell anything else that isn't unique or needs special handling
+					if(A.type in to_sell)
+						to_sell[A.type]["count"]++
+						qdel(A)	//We already have an object reference to use for values, duplicate items are safe to qdel
+					else
+						to_sell[A.type] = list("obj" = A, "count" = 1)
+
+			else
+				qdel(AM)
+
+	var/payout = 0
 
-			qdel(AM)
+	//Sell all items in bulk and qdel them
+	for(var/atom_type in to_sell)
```

Comment:
along with this

Reply:
Resolved in 28d597c3e46e3f3174ea47be272505b37db6c774
Check isn't needed, as the list won't be populated so nothing to iterate through